tests/rest/client/utils.py

```diff
@@ -196,6 +196,16 @@ def leave(self, room=None, user=None, expect_code=200, tok=None):
             expect_code=expect_code,
         )
 
+    def ban(self, room=None, src=None, targ=None, expect_code=200, tok=None):
```

Comment:
Type hints since we're adding this please! (And a docstring.)

Reply:
Ah, I wanted to hold off on this and change all of these helpers in one go. They all lack type hints, all incorrectly allow a bunch of required arguments to default to None, and a docstring. We could do better here by using `functools.partialmethod`.

Reply:
> We could do better here by using `functools.partialmethod`.

Ahh, that's not strictly true: for "join" and "leave" the src and target are one and the same.

Reply:
And PyCharm doesn't have good support for partial/partialmethod. See e.g. https://youtrack.jetbrains.com/issue/PY-18448 or https://youtrack.jetbrains.com/issue/PY-37275

Reply:
My thought was mostly that any new code should get type hints, even if we do intend to replace it in the future or something. Thanks for adding them!